### PR TITLE
sscs-10075 parties notified put implementation

### PR DESCRIPTION
**Pull request by dholaramgurjar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10075

### Change description ###

Implement PUT partiesNotified Hearings Endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/PartiesNotifiedMapping.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    public static PartiesNotified buildUpdatePartiesNotifiedPayload(HearingWrapper wrapper) {
+
+        PartiesNotified.PartiesNotifiedBuilder partiesNotifiedBuilder = PartiesNotified.builder();
+        partiesNotifiedBuilder.requestVersion(getVersionNumber(wrapper));
+        return partiesNotifiedBuilder.build();
+    }
+
+    public static String getHearingId(HearingWrapper wrapper) {
+        return Optional.of(wrapper)
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/PartiesNotifiedMapping.java:24]: You want to use ofNullable in these two

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+@RestController
+@FeignClient(name = "hmc-hearing", url = "${hmc.url}", configuration = FeignClientConfig.class)
+public interface HmcHearingPartiesNotifiedApi {
+
+    String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+
+    @PutMapping(value = "/partiesNotified", consumes = MediaType.APPLICATION_JSON_VALUE)
+    void updatePartiesNotifiedHearingRequest(
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java:22]: Any contract tests for this?

Comment (dholaramgurjar) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java:20]: I think @JamieHMCTS has written contract test , I am focusing on feign client call ,  Once merge both code they will come.

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/PartiesNotifiedMapping.java (hunk 1 above)
Comment (dholaramgurjar) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/PartiesNotifiedMapping.java:23]: Actually These are existing functions created for Post  end points, I am just reusing them.

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/PartiesNotifiedMapping.java]: Huh, I thought Optional.of would throw an NPE if anything was null in the chain

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/PartiesNotifiedMapping.java]: Huh, I thought Optional.of would throw an NPE if anything was null in the chain

**Pull request closed without merging**